pkg/slayers/path/scion/raw_spec.gobra

```diff
@@ -975,23 +958,23 @@ ensures
 	let prevSegLen := LengthOfPrevSeg(currHfIdx+1, seg1Len, seg2Len, seg3Len) in
 	let segLen := LengthOfCurrSeg(currHfIdx+1, seg1Len, seg2Len, seg3Len) in
 	let numInf := NumInfoFields(seg1Len, seg2Len, seg3Len) in
-	let offset := HopFieldOffset(numInf, 0, 0) in
-	CurrSeg(raw, offset + path.HopLen * prevSegLen, currInfIdx, currHfIdx-prevSegLen+1, segLen, 0) ==
+	let offset := HopFieldOffset(numInf, prevSegLen, 0) in
+	CurrSeg(raw, offset, currInfIdx, currHfIdx-prevSegLen+1, segLen, 0) ==
 		get(LeftSeg(raw, currInfIdx, seg1Len, seg2Len, seg3Len, 0))
 decreases
 func XoverCurrSeg(raw []byte, currInfIdx int, currHfIdx int, seg1Len int, seg2Len int, seg3Len int) {
```

Comment:
As a general comment, in future PRs, we should look for more opportunities to use `io.SegLens` (this function would be a good example). Please keep an eye for those.

Reply:
Yes, I believe there are multiple places where we can use `io.SegLens`. I will create a separate PR to address this.